Server/mods/deathmatch/logic/CConsoleCommands.cpp

```diff
@@ -1232,34 +1232,42 @@ bool CConsoleCommands::DebugScript(CConsole* pConsole, const char* szArguments,
         {
             CPlayer* pPlayer = static_cast<CPlayer*>(pClient);
 
-            // Convert to number
-            int iLevel = atoi(szArguments);
-            if (iLevel != (int)pPlayer->GetScriptDebugLevel())
+            // To prevent characters before converting
+            if (strcmp(szArguments, "0") == 0 || strcmp(szArguments, "1") == 0 || strcmp(szArguments, "2") == 0 || strcmp(szArguments, "3") == 0)
             {
-                // Between 0 and 3?
-                if (iLevel >= 0 && iLevel <= 3)
+                // Convert to number
+                int iLevel = atoi(szArguments);
```

Comment:
What if you put this here instead?

```suggestion
                int iLevel = atoi(szArguments);
                if (iLevel == 0 && strcmp(szArguments, "0") != 0) {
                    pEchoClient->SendEcho("debugscript: Syntax is 'debugscript <mode>'");
                    return false;
                }
```

We generally try to avoid adding loads of nested if statements if we can, but some of our old (like this!) doesn't do that.

Reply:
Done, reverted the old commit and added this instead.